wagtail/tests/utils.py

```diff
@@ -51,3 +60,122 @@ def reset_warning_registry():
         for mod in list(sys.modules.values()):
             if hasattr(mod, key):
                 getattr(mod, key).clear()
+
+
+class WagtailPageTests(WagtailTestUtils, TestCase):
+    """
+    A set of asserts to help write tests for your own Wagtail site.
+    """
+    def setUp(self):
+        super(WagtailPageTests, self).setUp()
+        self.login()
+
+    def _testCanCreateAt(self, parent_model, child_model):
+        child_ct = ContentType.objects.get_for_model(child_model)
+        parent_ct = ContentType.objects.get_for_model(parent_model)
+
+        return all([
+            child_ct in parent_model.allowed_subpage_types(),
+            # Anything can be created under a Page
+            parent_model is Page or parent_ct in child_model.allowed_parent_page_types()])
```

Comment:
Also, I don't think we need to check both `parent_model.allowed_subpage_types` and `child_model.allowed_parent_page_types` - either one of these will apply the rules from both sides.